server/app/controllers/admin/AdminProgramImageController.java

```diff
@@ -19,23 +23,51 @@ public final class AdminProgramImageController extends CiviFormController {
   private final ProgramService programService;
   private final ProgramImageView programImageView;
   private final RequestChecker requestChecker;
+  private final FormFactory formFactory;
 
   @Inject
   public AdminProgramImageController(
       ProgramService programService,
       ProgramImageView programImageView,
       RequestChecker requestChecker,
+      FormFactory formFactory,
       ProfileUtils profileUtils,
       VersionRepository versionRepository) {
     super(profileUtils, versionRepository);
     this.programService = checkNotNull(programService);
     this.programImageView = checkNotNull(programImageView);
     this.requestChecker = checkNotNull(requestChecker);
+    this.formFactory = checkNotNull(formFactory);
   }
 
   @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
   public Result index(Http.Request request, long programId) throws ProgramNotFoundException {
     requestChecker.throwIfProgramNotDraft(programId);
     return ok(programImageView.render(request, programService.getProgramDefinition(programId)));
   }
+
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result updateDescription(Http.Request request, long programId)
+      throws ProgramNotFoundException {
+    requestChecker.throwIfProgramNotDraft(programId);
+    Form<ProgramImageDescriptionForm> form =
+        formFactory
+            .form(ProgramImageDescriptionForm.class)
+            .bindFromRequest(
+                request, ProgramImageDescriptionForm.FIELD_NAMES.toArray(new String[0]));
+    String newDescription = form.get().getSummaryImageDescription();
+
+    programService.setSummaryImageDescription(
+        programId, LocalizedStrings.DEFAULT_LOCALE, newDescription);
+
+    String toastMessage;
+    if (newDescription.isBlank()) {
```

Comment:
Question for you actually: Is there a possibility that the description failed to save to the database for some reason? If so, is there an error/exception that I should be catching here and sending to the frontend?

Reply:
The only reason I could think of it failing is if there's a blip in the db or something. But I'm hesitant to show a success message without knowing that it saved. What do you think about checking for the `ProgramNotFoundException` that's thrown if it can't get the `programDefinition`?

Reply:
Yes! Done. I think `requestChecker.throwIfProgramNotDraft(programId)` might throw first because a non-existent program is also not a draft program? Maybe? But good to have both.